### PR TITLE
Handle OPTIONS HTTP requests correctly.

### PR DESCRIPTION
**Pull request by juliusv**
Fixes https://github.com/prometheus/prometheus/issues/1346

### CODE REVIEW

**Comment by juliusv** [comment]
@grobie @fabxc

**Comment by juliusv** [comment]
I added a test for the v1 API, but didn't bother to add one for the legacy one. Tested legacy manually though.

File: web/api/legacy/query.go [hunk 1]
```diff
@@ -61,6 +61,12 @@ func parseDuration(d string) (time.Duration, error) {
 	return time.Duration(dFloat * float64(time.Second/time.Nanosecond)), nil
 }
 
+// Options handles OPTIONS requests to /api/... endpoints.
```
Comment (grobie) [inline web/api/legacy/query.go]: Which /api/... endpoints?

Comment (juliusv) [inline web/api/legacy/query.go]: All of them?

Comment (grobie) [inline web/api/legacy/query.go]: Ah, I missed that we nest the legacy query methods under /api/ as well.

**Comment by grobie** [comment]
:+1:

File: web/api/v1/api.go [hunk 2]
```diff
@@ -56,10 +63,10 @@ type response struct {
 
 // Enables cross-site script calls.
 func setCORS(w http.ResponseWriter) {
-	w.Header().Set("Access-Control-Allow-Headers", "Accept, Authorization, Content-Type, Origin")
-	w.Header().Set("Access-Control-Allow-Methods", "GET")
-	w.Header().Set("Access-Control-Allow-Origin", "*")
-	w.Header().Set("Access-Control-Expose-Headers", "Date")
+	w.Header().Set("Access-Control-Allow-Headers", corsAllowedHeaders)
```
Comment (fabxc) [inline web/api/v1/api.go]: TBH, I don't quite get the reason for making the values const and then inlining the keys.
There's reuse in the tests, which guards against typos – but only for the values.

I would just make it a global `var corsHeaders map[string]string`. `setCors()` and the test can then simply use a loop over that.

Comment (juliusv) [inline web/api/v1/api.go]: Yeah, that's better. Updated.

**Comment by fabxc** [comment]
:+1:

**Pull request merged**